contrib/internal/telemetrytest/telemetry_test.go

```diff
@@ -0,0 +1,75 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+package telemetrytest
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/gorilla/mux"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+)
+
+// TestIntegrationInfo verifies that an integration leveraging instrumentation telemetry
+// sends the correct data to the telemetry client.
+func TestIntegrationInfo(t *testing.T) {
+	mux.NewRouter()
+	integrations := telemetry.Integrations()
+	assert.Len(t, integrations, 1)
+	assert.Equal(t, integrations[0].Name, "gorilla/mux")
+	assert.True(t, integrations[0].Enabled)
+}
+
+var TELEMETRY_IMPORT = "gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+
+type contribPkg struct {
+	ImportPath string
+	Name       string
+	Imports    []string
+}
+
+func (p *contribPkg) hasImport(imp string) bool {
+	for _, imp := range p.Imports {
+		if imp == TELEMETRY_IMPORT {
+			return true
+		}
+	}
+	return false
+}
+
+// TestTelemetryEnabled verifies that the expected contrib packages leverage instrumentation telemetry
+func TestTelemetryEnabled(t *testing.T) {
+	tracked := map[string]struct{}{"mux": {}}
+	cmd := "go list -json=ImportPath,Name,Imports  ../.././..."
+	body, err := exec.Command("bash", "-c", cmd).Output()
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	var packages []contribPkg
+	// need to reformat the output of the go list command to be a valid json
```

Comment:
I'm going to file an issue upstream that `go list -json` doesn't print valid json. That seems like a bug 😄

Reply:
In the meantime, you can link to [this issue](https://github.com/golang/go/issues/59440) in the comment (hopefully they'll eventually fix it!)

Reply:
Based on the issue I filed (which was closed), we should be able to do this instead: https://go.dev/play/p/KnmhmRnGmJt

i.e. it looks like we should be parsing this as a stream of distinct JSON values which is possible with https://pkg.go.dev/encoding/json#Decoder, rather than a single JSON list in a single `json.Unmarshal` call.

Reply:
Thanks for the solution! Just updated to use the json decoder